python/lsst/summit/utils/utils.py

```diff
@@ -1419,3 +1425,221 @@ def getSunAngle(time: Time | None = None) -> float:
+        alpha_band=0.2,
+        lw=2,
+        nbins=5,
+        is_scatter=False,
+    ):
+        """
+        Plot the best fit line, confidence interval,
+        and optionally scatter/binned data.
+
+        Parameters
+        ----------
+        ax : matplotlib.axes.Axes, optional
+            The axes to plot on. If None, uses current axes.
+        label : str, optional
+            Label for the best fit line.
+        color : str or None, optional
+            Color for the fit line and confidence band.
+        alpha_band : float, optional
+            Alpha transparency for the confidence interval band.
+        lw : int, optional
+            Line width for the best fit line.
+        nbins : int, optional
+            Number of bins for binned statistics.
+        is_scatter : bool, optional
+            If True, plot inliers and outliers as scatter points.
+
+        Returns
+        -------
+        ax : matplotlib.axes.Axes
+            The axes with the plot.
+        """
+        import matplotlib.pyplot as plt
+
+        if ax is None:
+            ax = plt.gca()
+
+        # Handle case where self.x contains only NaNs
+        if np.isnan(self.x).all():
+            raise ValueError("All x values are NaN; cannot plot best fit.")
+        xx = np.linspace(np.nanmin(self.x), np.nanmax(self.x), 200)
+        X_design = sm.add_constant(xx)
+
+        pred = self.ols_model.get_prediction(X_design)
+        summary_frame = pred.summary_frame(alpha=0.05)  # 95% intervals
+
+        mean = summary_frame["mean"]
+        ci_lo = summary_frame["mean_ci_lower"]
+        ci_hi = summary_frame["mean_ci_upper"]
+
+        # Plot best fit
+        ax.plot(xx, mean, color=color, label=label, lw=lw)
+        # Plot confidence interval band
+        ax.fill_between(xx, ci_lo, ci_hi, color=color, alpha=alpha_band)
+
+        mask = self.outlier_mask
+        xin, yin = self.x[~mask], self.y[~mask]  # inliers
+        xout, yout = self.x[mask], self.y[mask]  # outliers
+
+        if is_scatter:
+            # Plot inliers and outliers as before
+            ax.scatter(xin, yin, color=color or "black", alpha=0.7, label="Inliers", zorder=5)
+            ylims = ax.get_ylim()
+            ax.scatter(xout, yout, color="firebrick", alpha=0.5, label="Outliers", zorder=5)
+            ax.set_ylim(ylims)
+        # Bin data in nbins
+        if np.all(np.isnan(xout)):
+            # Skip binning if all values are NaN
+            pass
+        else:
+            bin_centers, means, stds, bin_counts = get_binned_data(xin, yin, nbins)
+
+            # Plot binned means and stds
+            ax.errorbar(
+                bin_centers,
+                means,
+                color=color or "black",
+                yerr=stds,
+                fmt="o",
+                capsize=4,
+                markersize=6,
+                alpha=0.7,
+                zorder=10,
+            )
+
+        return ax
+
+
+def get_binned_data(x, y, nbins):
```

Comment:
need mypy and docstring. Also, this is only needed by the fitter, so it might make more sense to put it inside the class. You can make it a @staticmethod to make it clear it doesn't use the state.